{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Review of retrograde model\n",
    "\n",
    "__Title:__ [The yeast retrograde response as a model of intracellular signaling of mitochondrial dysfunction](https://www.frontiersin.org/articles/10.3389/fphys.2012.00139/full)\n",
    "\n",
    "__Abstract__:\n",
    "Mitochondrial dysfunction activates intracellular signaling pathways that impact yeast\n",
    "longevity, and the best known of these pathways is the retrograde response. More recently,\n",
    "similar responses have been discerned in other systems, from invertebrates to human cells.\n",
    "However, the identity of the signal transducers is either unknown or apparently diverse,\n",
    "contrasting with the well-established signaling module of the yeast retrograde response.\n",
    "On the other hand, it has become equally clear that several other pathways and processes\n",
    "interact with the retrograde response, embedding it in a network responsive to a variety\n",
    "of cellular states. An examination of this network supports the notion that the master regulator\n",
    "NFκB aggregated a variety of mitochondria-related cellular responses at some point\n",
    "in evolution and has become the retrograde transcription factor. This has significant consequences\n",
    "for how we view some of the deficits associated with aging, such as inflammation.\n",
    "The support for NFκB as the retrograde response transcription factor is not only based on\n",
    "functional analyses. It is bolstered by the fact that NFκB can regulate Myc–Max, which is\n",
    "activated in human cells with dysfunctional mitochondria and impacts cellular metabolism.\n",
    "Myc–Max is homologous to the yeast retrograde response transcription factor Rtg1–Rtg3.\n",
    "Further research will be needed to disentangle the pro-aging from the anti-aging effects of\n",
    "NFκB. Interestingly, this is also a challenge for the complete understanding of the yeast\n",
    "retrograde response."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src='img/MT_retrograde/retrograde-pathway.jpg'>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# keynotes in retrograde response\n",
    "\n",
    "## [The yeast retrograde response as a model of intracellular signaling of mitochondrial dysfunction](http://www.frontiersin.org/Integrative_Physiology/10.3389/fphys.2012.00139/abstract)\n",
    "\n",
    "### Abstract\n",
    "\n",
    "1. Mitochondrial dysfunction activates intracellular signaling pathways that impact yeast longevity.\n",
    "    1. The best known one is the retrograde response\n",
    "    2. It is also discerned in invertebrates to human cells\n",
    "    3. NF$\\kappa$B is a retrograde response transcription factor.\n",
    "        1. discovered not only by functional analysis\n",
    "        2. NF$\\kappa$B can regulate Myc-Max, which is activated in human cells with dysfunctional mitochondria and impacts cellular metabolism\n",
    "2. Myc-Max is homologus to the yeast retrograde response transcription factor Rtg1-Rtg3.\n",
    "\n",
    "---\n",
    "\n",
    "### Introduction\n",
    "\n",
    "1. Mitochondrial dysfunction\n",
    "    1. Mitochondrial encephalomyopathy\n",
    "    2. cardiac hypertrophy\n",
    "    3. Parkinson's disease\n",
    "    4. cancer\n",
    "    5. May determine life span\n",
    "        1. *Caenorhabditis elegans*, *Drosophila melanogaster*, mouse, yeast\n",
    "2. Identify retrograde response in [bioinformatics approach](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2980572/pdf/nihms233811.pdf)\n",
    "3. Yeast retrograde response\n",
    "    1. $\\text{rho}^0$ vs. $\\text{rho}^{+}$\n",
    "        1. Lost TCA cycle $\\rightarrow$ lost glutamate synthesis, due to compromised activity of succinate dehydrogenase\n",
    "        2. Still poccess first three ractions\n",
    "            1. Metabolic adaptation: $\\alpha$-ketoglutarate $\\rightarrow$ glumate. If citrate is provided by the activation of glyoxylate cycle\n",
    "        3. No active electron transport chain\n",
    "4. Retrograde response\n",
    "    1. translocation of Rth1/3 heterodimer\n",
    "        1. helix-loop-helix/leucine zipper proteins\n",
    "        2. bind to the sequence GTCAC (R box)\n",
    "        3. Rtg1 is atypical, there is no apparent transcriptional activation domains\n",
    "        4. **Requires: Rtg2 protein**\n",
    "            1. Rtg2 has no known homologs in higher organism\n",
    "            2. Rtg2 promotes the dephosphorylation of Rtg3 by binding Mk1 and preventing Mk1 from forming a complex with the 14-4-4 protein Bmh1 or Bmh2 ([minor isoform](https://www.yeastgenome.org/locus/S000002506)), a complex which maintain in a hyperphosphorylated state.\n",
    "            3. Mks1 is removed by ubiquitin-mediated degradation promoted by the ubiquitin ligase component Grr1.\n",
    "                1. Grr1 $\\rightarrow$ positive regulator\n",
    "                2. Mks1 $\\rightarrow$ Negative regulator\n",
    "            4. TORC1 negatively regulate both upstream and downstream of Rtg2\n",
    "                1. Genetic studies:\n",
    "                    1. WD-protein Lst8/ mutation of Lst8\n",
    "                2. When glutamate and nutrient are plentiful, TORC1 downregulates pathway\n",
    "                3. TORC1 <-> negative feedback from dysfunctional mitochondria\n",
    "                    1. mediate phosphorylation of Sch9 (AGC protein kinase) is downregulated\n",
    "                    2. Phosorylated Sch9\n",
    "                        1. Antagonize stress responses\n",
    "                        2. \n",
    "            5. Reactions\n",
    "                1. Rtg2 + Mks1 $\\leftrightarrow$ Rtg2-Mks1\n",
    "                1. Mks1 + Bmh  $\\leftrightarrow$ Bmh-Mks1\n",
    "                1. Bmh-Mks1 + *-Rtg3$^{hyper-p_{i}}$ $\\leftrightarrow$ Bmh-Mks1-Rtg3$^{hyper-p_{i}}$ $\\rightarrow$ Bmh-Mks1 + Rtg3$^{partial-p_{i}}$\n",
    "                1. Mks1 + Grr1 $\\leftrightarrow$ Mks1-Grr1 $\\rightarrow$ Mks1$^{ubi}$ + Grr1\n",
    "                1. Mks1$^{ubi}$ $\\rightarrow$   \n",
    "\n",
    "                where [*-Rtg3$^{hyper-p_{i}}$] = Rtg3$^{hyper-p_{i}}$ + Rtg1-Rtg3$^{hyper-p_{i}}$\n",
    "    1. **only Rtg3 can bind to Rbox and activate transcription**\n",
    "    2. Rtg1 is in the top layer. abundant, long-lived, and noisy.\n",
    "    3. Osmotic stress can recruit Rtg1-Rtg3\n",
    "    4. [Ras2](https://www.yeastgenome.org/locus/S000005042): unclear, relate to cAMP-dependent pathway. Positive regulator\n",
    "        1. $\\uparrow$ lifespan via cAMP-dependent pathway\n",
    "    5. Rtg2 functions\n",
    "        1. Positive regulator of retrograde response\n",
    "            1. $\\uparrow$ dephosphorylation of Rtg3 in cytoplasm\n",
    "        2. Integral cofactor of transcriptional co-activator [SAGA-like complex](https://www.yeastgenome.org/go/GO:0046695), which is required for the induction of the retrograde response target gene CIT2, and it binds to CIT2 promoter\n",
    "        3. Genome stability (unknown mechanism)\n",
    "        4. **Sense dysfunctional mitochondria- possible mechanisms**\n",
    "            1. $\\downarrow$ of $\\Delta \\psi_{m}$ \n",
    "                1. When yeast ages, $\\Delta \\psi_m \\downarrow$, loses mtDNA, progressive activation of retrograde response.\n",
    "                2. Necessary and sufficient to induce retrograde response\n",
    "                3. how Rtg2 read $\\Delta \\psi_{m}$: unclear\n",
    "            2. ROS scavenger doesn't block signal\n",
    "            3. ATP$\\downarrow$ doesn't involve"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Reagents|function\n",
    "--------|--------\n",
    "Rtg1-Rtg3|activate retrograde response\n",
    "Rtg2|promote Rtg1-Rtg3: cytosol $\\rightarrow$ nucleus; binding with Mks1\n",
    "Mks1|phospholate Rtg3; downregulate RTG pathway\n",
    "Bmh1 or Bmh2|form Bmh1/2-Mks1 to phosphate Rtg3\n",
    "Grr1|degrade Mks1\n",
    "TORC1|negative regulator of the retrograde response; act both upstream and downstream of Rtg2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Review of retrograde signaling\n",
    "\n",
    "__Title:__ [Mitochondria to Nuclear Signaling Is Regulated by the Subcellular Localization of the Transcription Factors Rtg1p and Rtg3p](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC14906/pdf/mk002103.pdf)\n",
    "\n",
    "__Abstract__:\n",
    " Cells modulate the expression of nuclear genes in response to changes in the functional state of mitochondria, an interorganelle communication pathway called retrograde regulation. In yeast, expression of theCIT2 gene shows a typical retrograde response in that its expression is dramat-ically increased in cells with dysfunctional mitochondria, such as inro petites. Three genes control this  signaling  pathway:RTG1andRTG3 ,  which  encode  basic  helix-loop-helix  leucine  zipper transcription  factors  that  bind  as  heterodimer  to  theCIT2upstream  activation  site,  andRTG2 , which encodes a protein of unknown function. We show that in respiratory-competent (r1 ) cells in whichCIT2 expression is low, Rtg1p and Rtg3p exist as a complex largely in the cytoplasm, and inropetites in whichCIT2 expression is high, they exist as a complex predominantly localized in the  nucleus.  Cytoplasmic  Rtg3p  is  multiply  phosphorylated  and  becomes  partially  dephospho-rylated  when  localized  in  the  nucleus.  Rtg2p,  which  is  cytoplasmic  in  bothr1andro cells,  is required for the dephosphorylation and nuclear localization of Rtg3p. Interaction of Rtg3p with Rtg1p is required to retain Rtg3p in the cytoplasm ofr1 cells; in the absence of such interaction, nuclear localization and dephosphorylation of Rtg3p is independent of Rtg2p. Our data show that Rtg1p acts as both a positive and negative regulator of the retrograde response and that Rtg2p acts to transduce mitochondrial signals affecting the phosphorylation state and subcellular localization of Rtg3p\n",
    " \n",
    "<img src='img/MT_retrograde/rtg-delete.png'>\n",
    "<img src='img/MT_retrograde/translocate-hypo.png'>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Review of transcription facor tanslocation \n",
    "\n",
    "__Title:__ [Tunable Signal Processing Through Modular Control of Transcription Factor Translocation](http://science.sciencemag.org/content/339/6118/460.long)\n",
    "\n",
    "__Notes:__\n",
    "\n",
    "* Functional Domains for translocation signaling\n",
    "    * [Nucleus Export Sequence (NES)](https://en.wikipedia.org/wiki/Nuclear_export_signal)\n",
    "    * [Nucleus Localization Sequence (NLS)](https://en.wikipedia.org/wiki/Nuclear_localization_sequence)\n",
    "\n",
    "* Signal\n",
    "\n",
    "Functional Domain|P-phosphorylated|U-unphosphorylated\n",
    "---|---|---\n",
    "NES|$\\rightarrow$ Nucleus|$\\rightarrow$ Cytosol\n",
    "NLS|$\\rightarrow$ Cytosol|$\\rightarrow$ Nucleus\n",
    "\n",
    "* Mathematical model of translocation\n",
    "\n",
    "Because [there is only NLS site in RTG3](https://www.frontiersin.org/articles/10.3389/fphys.2012.00139/full), I only consider the case of _NLS only_ \n",
    "\n",
    "Ordinary differential equation:\n",
    "\n",
    "$$\\frac{dPr_{C}^{U}}{dt} = -(k_{in}^{'}+kp_{C})Pr_{C}^{U} + (k_{dp_C})Pr_{C}^{P} + (k_{out})Pr_{N}^{U} + 0 \\times Pr_{N}^{P}$$\n",
    "\n",
    "$$\\frac{dPr_{C}^{P}}{dt} = (k_{p_C})Pr_{C}^{U} - (k_{in} + k_{dp_C})Pr_{C}^{P} + 0\\times Pr_{N}^{U} +(k_{out})Pr_{N}^{P}$$\n",
    "\n",
    "$$\\frac{dPr_{N}^{U}}{dt} = (k_{in}^{'})Pr_{C}^{U}  + 0 \\times Pr_{C}^{P} - (k_{p_N}+k_{out}) Pr_{N}^{U} + (k_{dp_N}) Pr_{N}^{P}$$\n",
    "\n",
    "$$\\frac{dPr_{N}^{P}}{dt} = 0\\times Pr_{C}^{U} + (k_{in})Pr_{C}^{P} + (k_{p_N})Pr_{N}^{U} - (k_{out} + k_{dp_N})Pr_{N}^{P}$$"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Matrix form:\n",
    "\n",
    "\\begin{equation*}\n",
    "\\frac{d}{dt} \\begin{bmatrix} Pr_{C}^{U} \\\\ Pr_{C}^{P} \\\\ Pr_{N}^{U} \\\\ Pr_{N}^{P} \\end{bmatrix}\n",
    "=  \\begin{bmatrix}\n",
    "-(k_{in}^{'}+k_{p_C}) & k_{dp_C} & k_{out} & 0 \\\\\n",
    "k_{p_C} &  -(k_{in} + k_{dp_C})  & 0 & k_{out} \\\\\n",
    "k_{in}^{'}  & 0 & -(k_{p_N}+k_{out}) & k_{dp_N} \\\\\n",
    "0 & k_{in} & k_{p_N} & -(k_{out} + k_{dp_N}) \\\\\n",
    "\\end{bmatrix}\n",
    "\\begin{bmatrix} Pr_{C}^{U} \\\\ Pr_{C}^{P} \\\\ Pr_{N}^{U} \\\\ Pr_{N}^{P} \\end{bmatrix}\n",
    "\\end{equation*}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notation|Content\n",
    "---|---\n",
    "P|Phosphorylated\n",
    "U|Unphosphorylated\n",
    "C|cytoplasmic\n",
    "N|nuclear\n",
    "$Pr^{\\{U,P\\}}_{\\{C,N\\}}$|Protein with NLS only\n",
    "$k_{in}$|normal output rate for $Pr^{P}_{C}$\n",
    "$k_{in}^{'}$| enchanced input rate for $Pr^{U}_{C}$\n",
    "$k_{p_C}$| Cytoplasmic phosphorylation rate\n",
    "$k_{p_N}$| Nucleus phosphorylation rate\n",
    "$k_{dp_{C}}$| Cytoplasmic dephosphorylation rate\n",
    "$k_{dp_{N}}$| Nucleus dephosphorylation rate\n",
    "$k_{out}$|Export rate. Same value for NLS-only transcription factor"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src='img/MT_retrograde/translocate1.png'>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 4. RTG Network Analysis \n",
    "from [STRING](https://string-db.org/cgi/network.pl?taskId=bkkTcyhJN03C)\n",
    "\n",
    "__Conslusion:__\n",
    "\n",
    "1. Bottle nect effect in RTG-to-CIT2 pathway\n",
    "\n",
    "<img src='img/MT_retrograde/rtg-net.png'>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Review of Mitochondrial network scaling \n",
    "\n",
    "__Title:__ [Mitochondrial Network Size Scaling in Budding Yeast](http://science.sciencemag.org/content/338/6108/822)\n",
    "\n",
    "__Abstract:__ Mitochondria must grow with the growing cell to ensure proper cellular physiology and inheritance upon division. We measured the physical size of mitochondrial networks in budding yeast and found that mitochondrial network size increased with increasing cell size and that this scaling relation occurred primarily in the bud. The mitochondria–to–cell size ratio continually decreased in aging mothers over successive generations. However, regardless of the mother’sage or mitochondrial content, all buds attained the same average ratio. Thus, yeast populations achieve a stable scaling relation between mitochondrial content and cell size despite asymmetry in inheritance.\n",
    "\n",
    "__Conclusion:__\n",
    "\n",
    "1. Yeast mitochondrial volume = [0.3, 4] $\\mu m^{2}$\n",
    "\n",
    "<img src='img/MT_retrograde/yeast_mito_volume.jpg'>\n",
    "<img src='img/MT_retrograde/yeast_mito_volume2.png'>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ｍathematical Model of Retrograde Response\n",
    "\n",
    "## __Biochemical Network component__:\n",
    "    \n",
    "1. Input $\\vec{MT}$ with N species:\n",
    "$$\\vec{MT} := MT_1,...,MT_N$$\n",
    "where N is the total number of mitochondria\n",
    "\n",
    "1. Input coding:\n",
    " $$S = \\sum_{i=1}^{N} Volume_{MT_i} \\times Damaged_{MT_i}$$\n",
    " $$Volume_{MT_i} \\propto size_in_2D$$ \n",
    "   Distributions of $Volume_{MT_i}$ and $Damaged_{MT_i}$ can be derived from [Mitochondrial-netlogo-model](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0168198) \n",
    "   \n",
    "   1. $Volume_{MT_i} := \\{0, V_{max}\\}$\n",
    "       \n",
    "       $$V_{max} = \\frac{4}{3}h \\times Area_{MT_{i}}$$\n",
    "       \n",
    "       yeast mitochondria volime = [0.3, 4] $\\mu m^{2}$\n",
    "   \n",
    "   1. $Damaged_{MT_i} := \\{0, 1\\}$ where 0 represents healthy state, and 1 represents damaged state of mitochondrion i ($MT_i$)\n",
    "    \n",
    "2. Channel $\\nu$ with M species:\n",
    "   \n",
    "   1. Components:\n",
    "   $$V_1,...,V_M$$where M is the total number of proteins involved in network\n",
    "   \n",
    "   1. Label (in input-to-output order):\n",
    "       1. $V_{1}:=Rtg2_{C} ^{ina}$\n",
    "       1. $V_{2}:=Rtg2_{C} ^{act}$ \n",
    "       1. $V_{3}:=Rtg3_C^{P}$\n",
    "       1. $V_{4}:=Rtg3_C^{U}$\n",
    "       1. $V_{5}:=Rtg3_N^{P}$\n",
    "       1. $V_{6}:=Rtg3_N^{U}$\n",
    "       1. $V_{7}:=Rtg1_C$\n",
    "       1. $V_{8}:=Rtg1_N$\n",
    "       1. $V_{9}:={Rtg1-3}_C^{P}$\n",
    "       1. $V_{10}:={Rtg1-3}_C^{U}$\n",
    "       1. $V_{11}:={Rtg1-3}_N^{P}$\n",
    "       1. $V_{12}:={Rtg1-3}_N^{U}$\n",
    "       1. $V_{13}:=Bmh_{C}$\n",
    "       1. $V_{14}:= Bmh-Mks1_{C}$\n",
    "       1. $V_{15}:= Mks1-Rtg2^{ina}_{C}$\n",
    "       1. $V_{16}:= Mks1-Rtg2^{act}_{C}$\n",
    "       \n",
    "       __Note__: $Protein^{*}_{\\#}$, the notations represent *:{modification state}; #:{Location}\n",
    "  \n",
    "   2. $Rtg1-3_{\\#}^{*}$: hereodimer can activate CIT2 expression, then start retrograde response. \n",
    "       1. *: {p, u} (phosphtate, partially dephosphated)\n",
    "       1. #: {n,c} {nucleus, cytoplasm}\n",
    "   \n",
    "   3. [$Rtg2^{*}$](https://www.yeastgenome.org/locus/S000003221): \n",
    "       1. a sensor of mitochondrial dysfunction.\n",
    "       1. *: {inactivated (ina), activated (a)}\n",
    "   \n",
    "   3. TORC1 effect is ignored, for it is nutrient-dependent \n",
    "   \n",
    "   3. Ras2 is ignored and combined to the process of TORC1 inhibition\n",
    "\n",
    "3. Output $X$ with 1 componemt:\n",
    "   1. $X_{1}= {Rtg1/3}_{n}^{*}$\n",
    "   2. $[{Rtg1/3}_{n}^{*}]=[{Rtg1/3}_{n}^{p}]+[{Rtg1/3}_{n}^{u}]$"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Reaction\n",
    "\n",
    "The following displays all reactions mentioned in [review1](#Review-of-retrograde-model) and [review2](#Review-of-retrograde-signaling):\n",
    "\n",
    "RTG2 \n",
    "\n",
    "> $Rtg2_{C}^{ina} \\leftrightarrow Rtg2_{C}^{act}$ (Eq:1)\n",
    "\n",
    "RTG2, BMH AND MKS1\n",
    "\n",
    "> $Rtg2_{C}^{ina} + Mks1_{c} \\leftrightarrow Rtg2^{ina}-Mks1_{C}$ (Eq:2)\n",
    "\n",
    "> $Rtg2_{C}^{act} + Mks1_{C} \\leftrightarrow Rtg2^{act}-Mks1_{C}$ (Eq:3)\n",
    "\n",
    "> $Mks1 + Bmh  \\leftrightarrow Bmh-Mks1$ (Eq:4)\n",
    "\n",
    "> $Bmh-Mks1_{C} + Rtg3_{C}^{U} \\leftrightarrow Bmh-Mks1-Rtg3_{C}^{U} \\rightarrow$ $Bmh-Mks1_{C} + Rtg3_{C}^{P}$ (Eq:5)\n",
    "\n",
    "RTG3 \n",
    "\n",
    "> $Rtg3_{C}^{U} \\leftrightarrow Rtg3_{N}^{U}$ (Eq:6)\n",
    "\n",
    "> $Rtg3_{N}^{P} \\leftrightarrow Rtg3_{N}^{U}$ (Eq:7)\n",
    "\n",
    "RTG1 AND RTG3 \n",
    "\n",
    "> $Rtg3_{C}^{U} + Rtg1_{C} \\leftrightarrow Rtg1/3_{C}^{U}$ (Eq:8)\n",
    "\n",
    "> $Rtg3_{C}^{P} + Rtg1_{C} \\leftrightarrow Rtg1/3_{C}^{P}$ (Eq:9)\n",
    "\n",
    "> $Rtg3_{N}^{U} + Rtg1_{N} \\leftrightarrow Rtg1/3_{N}^{U}$ (Eq:10)\n",
    "\n",
    "> $Rtg3_{N}^{P} + Rtg1_{N} \\leftrightarrow Rtg1/3_{N}^{P}$ (Eq:11)\n",
    "\n",
    "RTG1/3 \n",
    "\n",
    "> $Rtg1/3_{C}^{U} \\leftrightarrow Rtg1/3_{N}^{U}$ (Eq:12)\n",
    "\n",
    "> $Rtg1/3_{C}^{P} \\leftrightarrow Rtg1/3_{N}^{P}$ (Eq:13)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Differential Equation-based Framework\n",
    "\n",
    "$$\\frac{d}{dt}[Rtg2_{C}^{ina}] = - k1 [Rtg2_{C}^{ina}] + kn1 [Rtg2^{act}_{C}]$$\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Necessary requirements:\n",
    "\n",
    "From [review2](#Review-of-retrograde-signaling), GFP experiments describes 14 localizton patterns by deletions of Rtg1, Rtg2, and Rtg3. The derived mathematical model has to fulltill all of the requirements that found in real world.\n",
    "\n",
    "\n",
    "\n",
    "1. Properties of Rtg3p-GFP localization \n",
    "\n",
    "Property|$[*-Rtg1_{*}]$| $[Rtg2_{C}]$ | S \n",
    "----------|--------|---------|---\n",
    "$[*-Rtg3^{*}_{C}] > [*-Rtg3^{*}_{N}]$| >0 | >0 | =0\n",
    "$[*-Rtg3^{*}_{C}] > [*-Rtg3^{*}_{N}]$| >0 | =0 | =0\n",
    "$[*-Rtg3^{*}_{C}] > [*-Rtg3^{*}_{N}]$| >0 | =0 | =high\n",
    "$[*-Rtg3^{*}_{C}] < [*-Rtg3^{*}_{N}]$| >0 | >0 | =high\n",
    "$[*-Rtg3^{*}_{C}] < [*-Rtg3^{*}_{N}]$| =0 | >0 | =0\n",
    "$[*-Rtg3^{*}_{C}] < [*-Rtg3^{*}_{N}]$| =0 | >0 | =high\n",
    "$[*-Rtg3^{*}_{C}] < [*-Rtg3^{*}_{N}]$| =0 | =0 | =0\n",
    "$[*-Rtg3^{*}_{C}] < [*-Rtg3^{*}_{N}]$| =0 | =0 | =high\n",
    "\n",
    "where \n",
    "$$[*-Rtg3^{*}_{C}]=[Rtg1-Rtg3^{P}_{C}] + [Rtg1-Rtg3^{U}_{C}] + [Rtg3^{U}_{C}] + [Rtg3^{P}_{C}]$$\n",
    "\n",
    "$$[*-Rtg3^{*}_{N}]=[Rtg1-Rtg3^{P}_{N}] + [Rtg1-Rtg3^{U}_{N}] + [Rtg3^{U}_{N}] + [Rtg3^{P}_{N}]$$\n",
    "\n",
    "$$[*-Rtg1_{*}] = [Rtg1_{C}] + [Rtg1_{N}] + [Rtg1-Rtg3^{P}_{C}] + [Rtg1-Rtg3^{U}_{C}] + [Rtg1-Rtg3^{P}_{N}] + [Rtg1-Rtg3^{U}_{N}]$$\n",
    "\n",
    "2. Properties of Rtg1p-GFP localization\n",
    "\n",
    "Properties|$[Rtg3^{*}_{\\#}]$| $[Rtg2_{C}]$ | S \n",
    "----------|--------|---------|---\n",
    "$[*-Rtg1_{C}]>[*-Rtg1_{N}]$|>0|>0|=0\n",
    "$[*-Rtg1_{C}]>[*-Rtg1_{N}]$|>0|=0|=0\n",
    "$[*-Rtg1_{C}]>[*-Rtg1_{N}]$|>0|=0|=high\n",
    "$[*-Rtg1_{C}]>[*-Rtg1_{N}]$|=0|>0|=0\n",
    "$[*-Rtg1_{C}]>[*-Rtg1_{N}]$|=0|>0|=high\n",
    "$[*-Rtg1_{C}]<[*-Rtg1_{N}]$|>0|>0|=high\n",
    "\n",
    "where\n",
    "\n",
    "$$[*-Rtg1_{C}] = [Rtg1_{C}] + [Rtg1-Rtg3^{P}_{C}] + [Rtg1-Rtg3^{U}_{C}]$$\n",
    "\n",
    "$$[*-Rtg1_{N}] = [Rtg1_{N}] + [Rtg1-Rtg3^{P}_{N}] + [Rtg1-Rtg3^{U}_{N}]$$\n",
    "\n",
    "$$[Rtg3^{*}_{\\#}] = [*-Rtg3^{*}_{C}] + [*-Rtg3^{*}_{N}]$$"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Truth Table and Karnaugh Map\n",
    "\n",
    "In order to display these properties, I used boolean funtion to model input-output relation, and applied [Karnaugh map](https://en.wikipedia.org/wiki/Karnaugh_map) to find the simplest boolean formula of RTG pathway.\n",
    "\n",
    "\n",
    "### 1. Rtg3p Translocation\n",
    "\n",
    "__Definition__\n",
    "\n",
    "Condition|Mapping to Boolean Space\n",
    "---|---\n",
    "$[*-Rtg3^{*}_{C}] < [*-Rtg3^{*}_{N}]$ | 1\n",
    "$[*-Rtg3^{*}_{C}] > [*-Rtg3^{*}_{N}]$ | 0\n",
    "$[*-Rtg1_{*}]>0$ | 1\n",
    "$[*-Rtg1_{*}]=0$| 0\n",
    "$[Rtg2_{C}]>0$ | 1\n",
    "$[Rtg2_{C}]=0$ | 0\n",
    "$S=high$|1\n",
    "$S=0$|0\n",
    "\n",
    "\\begin{equation}\n",
    "Translocation_{Rtg3} := \\left\\{\\begin{matrix}\n",
    "1 \\text{ if } [*-Rtg3^{*}_{C}] < [*-Rtg3^{*}_{N}]\\\\ \n",
    "0 \\text{ if }[*-Rtg3^{*}_{C}] > [*-Rtg3^{*}_{N}]\n",
    "\\end{matrix}\\right.\n",
    "\\end{equation}\n",
    "\n",
    "__Truth Table:__\n",
    "\n",
    "$Bool([*-Rtg1_{*}])$ | $Bool([Rtg2_{C}])$ | S | $Translocation_{Rtg3}$ (Output)\n",
    "---|---|---|---\n",
    "0|0|0|x\n",
    "0|0|1|1\n",
    "0|1|0|1\n",
    "0|1|1|1\n",
    "1|0|0|0\n",
    "1|0|1|0\n",
    "1|1|0|0\n",
    "1|1|1|1\n",
    "\n",
    "__Result__:\n",
    "\n",
    "$$Translocation_{Rtg3} = \\sim Bool([*-Rtg1_{*}]) + Bool([Rtg2_{C}])\\times S$$\n",
    "\n",
    "This model expects (0,0,0,1)\n",
    "\n",
    "\n",
    "### 2. Rtg1p Translocation\n",
    "\n",
    "__Definition__\n",
    "    \n",
    "Condition|Mapping to Boolean Space\n",
    "---|---\n",
    "$[*-Rtg1_{C}]>[*-Rtg1_{N}]$|0\n",
    "$[*-Rtg1_{C}]<[*-Rtg1_{N}]$|1\n",
    "$[Rtg3^{*}_{\\#}]>0$|1\n",
    "$[Rtg3^{*}_{\\#}]=0$|0\n",
    "$[Rtg2_{C}]>0$|1 \n",
    "$[Rtg2_{C}]=0$|0 \n",
    "$S=high$|1\n",
    "$S=0$|0\n",
    "\n",
    "\\begin{equation}\n",
    "Translocation_{Rtg1} := \\left\\{\\begin{matrix}\n",
    "1 \\text{ if } [*-Rtg1_{C}]<[*-Rtg1_{N}]\\\\ \n",
    "0 \\text{ if }[*-Rtg1_{C}]>[*-Rtg1_{N}]\n",
    "\\end{matrix}\\right.\n",
    "\\end{equation}\n",
    "\n",
    "__Truth table__\n",
    "\n",
    "$Bool([Rtg3^{*}_{\\#}])$ | $Bool([Rtg2_{C}])$ | S | $Translocation_{Rtg1}$ (Output)\n",
    "---|---|---|---\n",
    "0|0|0|x\n",
    "0|0|1|x\n",
    "0|1|0|0\n",
    "0|1|1|0\n",
    "1|0|0|0\n",
    "1|0|1|0\n",
    "1|1|0|0\n",
    "1|1|1|1\n",
    "\n",
    "\n",
    "where x is \"don't care term\", for some relations aren't verified in [review2](#Review-of-retrograde-signaling)\n",
    "\n",
    "\n",
    "__Result__:\n",
    "\n",
    "$$Transduction_{Rtg1} = Bool([Rtg3^{*}_{\\#}]) \\times Bool([Rtg2_{C}]) \\times S$$\n",
    "\n",
    "This model expects (0,0,0,0), (0,0,1,0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Construct a channel \n",
    "\n",
    "$$x_1 = channel(\\vec{MT}, \\vec{V})$$\n",
    "\n",
    "where $x_1 := Rtg1/3_{N}^{*} = Rtg1/3_{N}^{U}+ Rtg1/3_{N}^{P}$ "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Define \n",
    "$$Hill_act(s_{i})=\\frac{}{}$$"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Question: How to achieve maximum information transmission under known $\\vec{MT}$__:\n",
    "\n",
    "The model $$[MT_1, MT_2,...,MT_N]->S_1->\\nu->X_1$$ In order to decode $X_1$, the ouput needs to one-to-one mapping to all possibilities of input array, that is, $\\vec{MT}$, there are $2^{N}$ possible combination. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Frequency Response\n",
    "\n",
    "__Lemma__:\n",
    "\n",
    "A system can be linearized by the following equations:\n",
    "\n",
    "$$\\frac{d}{dt}\\textbf{x} = f(\\textbf{x}(t),\\textbf{u}(t)) = \\textbf{A}x(t) + \\textbf{B}u(t) $$\n",
    "\n",
    "$$y(t) = h(\\textbf{x}(t),\\textbf{u}(t)) = \\textbf{C}x(t) + \\textbf{D}u(t)$$\n",
    "\n",
    "where \n",
    "\n",
    "Symbol|Content|Property\n",
    "---|---|---\n",
    "$\\textbf{x}$|System state|Vector\n",
    "$\\textbf{u}$|Input|Vector\n",
    "y|output|value\n",
    "$\\textbf{A}$|System Jacobian|Matrix\n",
    "$\\textbf{B}$|Input maps|Matrix\n",
    "$\\textbf{C}$|Output Maps|Matrix\n",
    "$\\textbf{D}$|Feed-forward term|Matrix\n",
    "\n",
    "Then\n",
    "\n",
    "$$H(\\omega) = \\textbf{C}(i\\omega\\textbf{I}-\\textbf{A})^{-1}\\textbf{B} + D$$\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
